pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -354,6 +354,27 @@ func (r *RestartReconciler) removePodsWithClusterUpState(ctx context.Context, po
 	return pods[:i], ctrl.Result{}, nil
 }
 
+// doesDBHaveQuorum returns true if more than half
+// of the primary nodes are up. Only nodes that are
+// already part of the database are considered
+func (r *RestartReconciler) doesDBHaveQuorum(clusterState map[string]string) bool {
+	totalPrimaryCount := 0
+	upPrimaryCount := 0
+	for _, pod := range r.PFacts.Detail {
+		if !pod.isPrimary {
+			continue
+		}
+		state, ok := clusterState[pod.vnodeName]
+		if ok {
+			totalPrimaryCount++
+			if state == vadmin.StateUp {
```

Comment:
I think we need to use the clusterState to know if a node exists in the catalog, but we should use the podfacts for up/down. We can come to the conclusion a bit quicker than spread does. So, it can be likely to see a pod that is down but the cluster state says is up.